contrib/cindercompatibleapi/api/router.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements a entry into the OpenSDS northbound REST service.
+
+*/
+
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/astaxie/beego"
+	bctx "github.com/astaxie/beego/context"
+	c "github.com/opensds/opensds/client"
+	"github.com/opensds/opensds/contrib/cindercompatibleapi/converter"
+	"github.com/opensds/opensds/pkg/api/filter/auth"
+	"github.com/opensds/opensds/pkg/api/filter/context"
+	"github.com/opensds/opensds/pkg/utils/constants"
+)
+
+const (
+	// StatusOK ...
+	StatusOK = http.StatusOK
+	// StatusAccepted ...
+	StatusAccepted = http.StatusAccepted
+
+	// OpensdsEndpoint ...
+	OpensdsEndpoint = "OPENSDS_ENDPOINT"
+	// OpensdsAuthStrategy ...
+	OpensdsAuthStrategy = "OPENSDS_AUTH_STRATEGY"
+	// OpensdsTenantID ...
+	OpensdsTenantID = "OPENSDS_TENANT_ID"
+
+	// OsAuthURL ...
+	OsAuthURL = "OS_AUTH_URL"
+	// OsUsername ...
+	OsUsername = "OS_USERNAME"
+	// OsPassword ...
+	OsPassword = "OS_PASSWORD"
+	// OsTenantName ...
+	OsTenantName = "OS_TENANT_NAME"
+	// OsProjectName ...
+	OsProjectName = "OS_PROJECT_NAME"
+	// OsUserDomainID ...
+	OsUserDomainID = "OS_USER_DOMAIN_ID"
+)
+
+var (
+	client *c.Client
+)
+
+// Run ...
+func Run(CinderEndPoint string) {
+	ep, ok := os.LookupEnv(OpensdsEndpoint)
```

Comment:
It's better if these parameters (OpensdsEndpoint and OpensdsAuthStrategy) can be passed from console by using `flag`.

Reply:
Using flag will changed from reading an auto-read variable to reading a variable manually, which in turn will complicate the operation.So it cannot be changed here.